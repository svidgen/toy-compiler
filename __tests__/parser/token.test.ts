import {
	token,
	Token
} from '../../src/parser';

describe('parser', () => {
	describe('Tokens', () => {

		it('can parse a keyword using string', () => {
			const T = token('T', 'whatever');
			const code = 'whatever';
			const ast = T.parse({code});
			expect(ast).toMatchSnapshot();
		});

		it('can parse a keyword using regex', () => {
			const T = token('T', /whatever/);
			const code = 'whatever';
			const ast = T.parse({code});
			expect(ast).toMatchSnapshot();
		});

		it('returns `null` on a non-match using string', () => {
			const T = token('T', 'whatever');
			const code = 'not whatever';
			const ast = T.parse({code});
			expect(ast).toEqual(null);
		});

		it('returns `null` on a non-match using regex', () => {
			const T = token('T', /whatever/);
			const code = 'not whatever';
			const ast = T.parse({code});
			expect(ast).toEqual(null);
		});

	});
});
