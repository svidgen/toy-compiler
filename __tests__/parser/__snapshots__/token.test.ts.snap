// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser Tokens can parse a keyword using regex 1`] = `
AST {
  "children": [],
  "code": "whatever",
  "end": 8,
  "start": 0,
  "type": "T",
}
`;

exports[`parser Tokens can parse a keyword using string 1`] = `
AST {
  "children": [],
  "code": "whatever",
  "end": 8,
  "start": 0,
  "type": "T",
}
`;
